test/integration/integration_admin_test.cc

```diff
@@ -1,133 +1,139 @@
 
   // Bad level
   response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/logging?level=blah",
-                                                "", Http::CodecClient::Type::HTTP1);
+                                                "", Http::CodecClient::Type::HTTP1, GetParam());
   EXPECT_TRUE(response->complete());
   EXPECT_STREQ("404", response->headers().Status()->value().c_str());
 
   // Bad logger
   response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/logging?blah=info",
-                                                "", Http::CodecClient::Type::HTTP1);
+                                                "", Http::CodecClient::Type::HTTP1, GetParam());
   EXPECT_TRUE(response->complete());
   EXPECT_STREQ("404", response->headers().Status()->value().c_str());
 
   // This is going to stomp over custom log levels that are set on the command line.
-  response = IntegrationUtil::makeSingleRequest(
-      lookupPort("admin"), "GET", "/logging?level=warning", "", Http::CodecClient::Type::HTTP1);
+  response =
+      IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/logging?level=warning", "",
+                                         Http::CodecClient::Type::HTTP1, GetParam());
   EXPECT_TRUE(response->complete());
   EXPECT_STREQ("200", response->headers().Status()->value().c_str());
   for (const Logger::Logger& logger : Logger::Registry::loggers()) {
     EXPECT_EQ("warning", logger.levelString());
   }
 
   response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/logging?assert=trace",
-                                                "", Http::CodecClient::Type::HTTP1);
+                                                "", Http::CodecClient::Type::HTTP1, GetParam());
   EXPECT_TRUE(response->complete());
   EXPECT_STREQ("200", response->headers().Status()->value().c_str());
   EXPECT_EQ(spdlog::level::trace, Logger::Registry::getLog(Logger::Id::assert).level());
 
   const char* level_name = spdlog::level::level_names[default_log_level_];
   response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET",
                                                 fmt::format("/logging?level={}", level_name), "",
-                                                Http::CodecClient::Type::HTTP1);
+                                                Http::CodecClient::Type::HTTP1, GetParam());
   EXPECT_TRUE(response->complete());
   EXPECT_STREQ("200", response->headers().Status()->value().c_str());
   for (const Logger::Logger& logger : Logger::Registry::loggers()) {
     EXPECT_EQ(level_name, logger.levelString());
   }
 }
 
-TEST_F(IntegrationTest, Admin) {
+TEST_P(IntegrationAdminTest, Admin) {
   BufferingStreamDecoderPtr response = IntegrationUtil::makeSingleRequest(
-      lookupPort("admin"), "GET", "/", "", Http::CodecClient::Type::HTTP1);
+      lookupPort("admin"), "GET", "/", "", Http::CodecClient::Type::HTTP1, GetParam());
```

Comment:
Can we add a wrapper method for `GetParam()` that provides semantic information about what the parameter is? E.g. `GetIpVersion()`?